scripts/host-check

```diff
@@ -1169,6 +1181,88 @@ def check_shmem_pages_max_size() -> CheckFuncReturn:
     return CheckState.SUCCESS, f"{shared_mem_max_page_size:.1f} GiB"
 
 
+def _module_params_compare(
+    module: str, exp_param_vals: Dict[str, str]
+) -> Dict[str, Optional[str]]:
+    """
+    For a kernel module, check if the runtime parameters have the expected
+    values.
+
+    :param module:
+        Name of the module
+
+    :param exp_param_vals:
+        Dictionary mapping parameter name to the expected value of the parameter
+
+    :return:
+        A dict containing parameters with unexpected values
+        If failed to check value for a parameter, returns None as the dict
+        value for the parameter
+    """
+    param_unexpected: Dict[str, Optional[str]] = {}
+
+    if not _is_module_loaded(
+        module.replace("-", "_")
+    ) and not _is_module_builtin(module):
+        # Skip module param check if module is not loaded
+        return param_unexpected
+    # sysfs filesystem requires the underscore version of the kernel module
+    # name
+    module = module.replace("-", "_")
+    all_params_str, _ = run_cmd(["modinfo", "--field", "parm", module])
+    for param, exp_val in exp_param_vals.items():
+        if not re.search(rf"^{param}:", all_params_str, re.MULTILINE):
+            # if 'param' is absent in modinfo output, it means either the
+            # parameter was introduced in a later kernel version, or the
+            # specific linux distribution does not support the parameter.
+            continue
+        try:
+            path = f"/sys/module/{module}/parameters/{param}"
+            with open(path, "r", encoding="utf-8") as f:
+                param_val = f.read().strip()
+            # (null) implies the parameter isn't set, and is thus the
```

Comment:
I think we need a comment somewhere which highlights that at the moment we're only checking for default values (and so treating (null) as success) - if we added a check for non-default values, then the logic here would need changing.

Reply:
Ahh yeah true. Would changing `MODULE_EXPECTED_PARAMS` be better:
```
    "igb_uio": {
        "intr_mode": {"msix", "null"},
    },
```

Reply:
Yes, that seems more robust. Is 'intr_mode' the only parameter that can be not-set? i.e. this doesn't apply for the vfio_pci parameters?

Reply:
Yes, looks like that's correct for boolean parameters - so what you've got looks good.